ocs_ci/ocs/ocp.py

```diff
@@ -376,3 +376,42 @@ def switch_to_default_rook_cluster_project():
         bool: True on success, False otherwise
     """
     return switch_to_project(defaults.ROOK_CLUSTER_NAMESPACE)
+
+
+def get_all_nodes():
+    """
+    Get all cluster nodes
+
+    Returns:
+        list: Cluster node dictionaries
+
+    """
+    nodes_obj = OCP(kind='node')
+    return nodes_obj.get()['items']
+
+
+def wait_for_nodes_ready(num_of_nodes):
+    """
+    Wait until all nodes are in Ready status
+
+    Args:
+        num_of_nodes (int): The number of nodes to wait for
+
+    Returns:
+        bool: True if all nodes reached status Ready
+
+    """
+    try:
+        for sample in TimeoutSampler(timeout=120, sleep=3, func=get_all_nodes):
+            if len(sample) == num_of_nodes:
+                ready_nodes = 0
+                for node in sample:
+                    for con in node.get('status').get('conditions'):
+                        if 'True' in con.get('status'):
+                            if con.get('type') == 'Ready':
+                                ready_nodes += 1
+                    if ready_nodes == num_of_nodes:
+                        return True
+    except TimeoutExpiredError:
+        log.error("Not all nodes reached status Ready")
```

Comment:
@ebenahar will it be possible to display exactly which all nodes didn't reach Ready state ? Just a question

Reply:
Thanks, I'll add this as part of https://github.com/red-hat-storage/ocs-ci/issues/504